### PR TITLE
Port storage and media tests to use devtools

### PR DESCRIPTION
**Pull request by v-iam**
This branch adds a dependency on https://github.com/Azure/azure-python-devtools and ports the storage and media tests to use it for their resource preparation and recording setup/teardown.

As well, it migrates those tests to use the new directory structure, with tests alongside the modules they test instead of lumped together in a single directory. But the tests in that old directory (`azure-mgmt/tests`) should still work.

The new integration layer around azure-python-devtools is in the top-level `azure-sdk-testutils` directory.

### CODE REVIEW

File: azure-mgmt/tests/test_batch.py [hunk 1]
```diff
@@ -215,7 +215,8 @@ def setUpClass(cls):
             cls.batch_client_sk, cls.batch_client_aad = create_batch_account(cls.batch_mgmt_client, cls.settings, cls.live)
         except Exception as err:
             cls.tearDownClass()
-            raise AssertionError("Failed to setup Batch Account: {}".format(err))
+            raise
```
Comment (lmazuel) [inline azure-mgmt/tests/test_batch.py]: What is this doing here?

Comment (v-iam) [inline azure-mgmt/tests/test_batch.py]: Oops?

IIRC I hit this at some point and the original raise was obscuring the actual exception. I'll revert it.

**Comment by v-iam** [comment]
I think the problem is I needed to add the new directories with tests in them to `setup.cfg`. I did so in the last commit and it seems like the tests are recognized now, but they also broke against 2.7 for some reason.

**Comment by lmazuel** [comment]
@v-iam Py2.7 and PyPy (which is 2.7 syntax) are failing for the same reason, so I guess there is some Py3 only syntax somewhere :)

**Comment by v-iam** [comment]
@lmazuel LMK if you need anything else in order to get this merged. I made some changes since you approved it, notably:
* the "compatibility layer" for the test SDK is now a package, `azure-sdk-testutils/devtools_testutils`. This fixed the python2.7 CI error, though I don't know why, but it also matches better how the CLI test layer works. Prior to this the wrappers around azure-devtools were just a bunch of files in a folder; I think it's nicer and more consistent to have them be a package.
* I bumped the azure-devtools dependency to 0.4.1. This is important because [it fixes a bug](https://github.com/Azure/azure-python-devtools/pull/8) that may have caused tests to be skipped erroneously (which might account for the slight coverage decrease that was here previously).

**Comment by v-iam** [comment]
Updated to port ACR tests as well.

**Pull request merged**